browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.ts

```diff
@@ -433,6 +448,10 @@ export class SettingsBraveAddWalletNetworkDialogElement extends SettingsBraveAdd
         this.showCurrencyError()
         return;
       }
+      else if(nativeCurrency.decimals < 0) {
+        this.showNegativeDecimalsError()
```

Comment:
Let's also name it `showIncorrectDecimalsError` and add a check for a valid integer, i.e. decimals should not have decimals :)